src/main/src/main/java/org/geoserver/security/password/RandomPasswordProvider.java

```diff
@@ -44,6 +44,13 @@ public class RandomPasswordProvider {
      */
     public static int DefaultPasswordLength = 40;
 
+    /**
+     * Cryptographically strong random number generator. As a subclass of {@link java.util.Random},
+     * it's guaranteed to be thread safe (although not necessarily scalable, but for this use case
+     * it should not be a problem).
+     */
+    SecureRandom random = new SecureRandom();
```

Comment:
Is this driven by new rules in 4.7.3, or just coincidental?

Reply:
Driven by new rules, quoting:

> ### DMI: Random object created and used only once (DMI_RANDOM_USED_ONLY_ONCE)
> 
> This code creates a java.util.Random object, uses it to generate one random number, and then discards the Random object. This produces mediocre quality random numbers and is inefficient. If possible, rewrite the code so that the Random object is created once and saved, and each time a new random number is required invoke a method on the existing Random object to obtain it.
> 
> If it is important that the generated Random numbers not be guessable, you must not create a new Random for each random number; the values are too easily guessable. You should strongly consider using a java.security.SecureRandom instead (and avoid allocating a new SecureRandom for each random number needed).